ts/features/payments/checkout/screens/WalletPaymentDetailScreen.tsx

```diff
@@ -251,10 +251,19 @@ const WalletPaymentDetailContent = ({
     formatNumberAmount(amountValue, true, "right")
   );
 
+  const YEARS_TO_EXPIRE = 10;
+
   const dueDate = pipe(
     payment.dueDate,
     O.fromNullable,
-    O.map(_ => format(_, "DD/MM/YYYY")),
+    O.map(date => {
+      const formattedDate = format(date, "DD/MM/YYYY");
+      const tenYearsFromNow = new Date();
+      tenYearsFromNow.setFullYear(
+        tenYearsFromNow.getFullYear() + YEARS_TO_EXPIRE
+      );
+      return new Date(date) > tenYearsFromNow ? undefined : formattedDate;
+    }),
```

Comment:
What do you think about extracting this logic into a utility function within the `ts/features/payments/checkout/utils/index.ts` file and adding tests to cover the various use cases of the function?

Reply:
Addressed in this commit 4ec9fc3fae57798db56dc37dca0166278ea2d0a2